include/oneapi/dpl/internal/scan_by_segment_impl.h

```diff
@@ -0,0 +1,374 @@
+/* Copyright (c) 2023 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ *  
+ * Copyright (c) 2013, NVIDIA CORPORATION.  All rights reserved.
```

Comment:
I am worried about this licensing. Let's discuss why it's Nvidia stuff and probably rewrite it by our own